sys/senml/senml.c

```diff
@@ -0,0 +1,416 @@
+            }
+
+            ins_pos += chars_written;
+            buf_len -= chars_written;
+        }
+
+        if (curr_record->unit) {
+            chars_written = snprintf(&output[ins_pos], buf_len, "\"u\":\"%s\",",
+                                     curr_record->unit);
+
+            if (chars_written >= (ssize_t)buf_len || chars_written < 0) {
+                return -1;
+            }
+
+            ins_pos += chars_written;
+            buf_len -= chars_written;
+        }
+
+        if (curr_record->time != 0) {
+            chars_written = snprintf(&output[ins_pos], buf_len, "\"t\":%f,",
+                                     curr_record->time);
+
+            if (chars_written >= (ssize_t)buf_len || chars_written < 0) {
+                return -1;
+            }
+
+            ins_pos += chars_written;
+            buf_len -= chars_written;
+        }
+
+        if (curr_record->update_time != 0) {
+            chars_written = snprintf(&output[ins_pos], buf_len, "\"ut\":%f,",
+                                     curr_record->update_time);
+
+            if (chars_written >= (ssize_t)buf_len || chars_written < 0) {
+                return -1;
+            }
+
+            ins_pos += chars_written;
+            buf_len -= chars_written;
+        }
+
+        if (curr_record->value_sum != 0) {
+            chars_written = snprintf(&output[ins_pos], buf_len, "\"s\":%f,",
+                                     curr_record->value_sum);
+
+            if (chars_written >= (ssize_t)buf_len || chars_written < 0) {
+                return -1;
+            }
+
+            ins_pos += chars_written;
+            buf_len -= chars_written;
+        }
+
+        if (curr_record->value_type != SENML_TYPE_UNDEF) {
+            if (curr_record->value_type == SENML_TYPE_FLOAT) {
+                chars_written = snprintf(&output[ins_pos], buf_len, "\"v\":%f,",
+                                         curr_record->value.f);
+            }
+            else if (curr_record->value_type == SENML_TYPE_STRING) {
+                chars_written = snprintf(&output[ins_pos], buf_len, "\"vs\":\"%s\",",
+                                         curr_record->value.s);
+            }
+            else if (curr_record->value_type == SENML_TYPE_BOOL) {
+                chars_written = snprintf(&output[ins_pos], buf_len, "\"vb\":%s,",
+                                         curr_record->value.b ? "true" : "false");
+            }
+            else if (curr_record->value_type == SENML_TYPE_BINARY) {
+                chars_written = snprintf(&output[ins_pos], buf_len, "\"vd\":\"%s\",",
+                                         curr_record->value.d);
+            }
+
+            if (chars_written >= (ssize_t)buf_len || chars_written < 0) {
+                return -1;
+            }
+
+            ins_pos += chars_written;
+            buf_len -= chars_written;
+        }
+
+        if (buf_len == buf_len_old) {   // record was empty, no values inserted: remove '{'
```

Comment:
No, the '{' does get inserted, check [line 274 in senml.c](https://github.com/RIOT-OS/RIOT/pull/5544/files#diff-17dca0f32589a71ffb5cf494b254ecdaR274), and the test application. It includes documents without base info.

The check for empty records was included with the following scenario in mind:
Suppose you initialize an array of `senml_record_t` of length 4, but then (for some reason) you can only fill 2 or 3 of the records. `pack.num` will still be 4, but the resulting JSON document will only contain records with actual content (the '{' for the empty records will be removed).